aodndata/moorings/products_handler.py

```diff
@@ -0,0 +1,98 @@
+import json
+
+from owslib.fes import PropertyIsEqualTo, PropertyIsNotEqualTo, PropertyIsLike, PropertyIsNotEqualTo, And
+
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType, FileType, PipelineFileCollection, PipelineFile
+from aodncore.pipeline.exceptions import ComplianceCheckFailedError, InvalidFileContentError, InvalidFileNameError
+from aodncore.pipeline.files import RemotePipelineFileCollection
+from aodncore.util.wfs import ogc_filter_to_string
+
+from aodndata.moorings.classifiers import MooringsFileClassifier
+
+
+class MooringsProductClassifier(MooringsFileClassifier):
+    @classmethod
+    def _get_data_category(cls, input_file):
+        return 'aggregated_timeseries'
+
+    @classmethod
+    def _get_product_level(cls, input_file):
+        return ''
+
+
+class MooringsProductsHandler(HandlerBase):
+    """Handler to create products from moorings files.
+
+    The input file is a JSON document containing a site_code and a list of variables. The handler will then create
+    the products for each variable at that site, using all the relevant input files available on S3.
+
+    The structure of the JSON document will be something like this:
+    {
+        "site_code": "NRSMAI",
+        "variables": ["TEMP", "PSAL", "DOX1", "DOX2", "CPHL"]
+    }
+    """
+
+    FILE_INDEX_LAYER = 'imos:moorings_all_map'
+
+    def __init__(self, *args, **kwargs):
+        super(MooringsProductsHandler, self).__init__(*args, **kwargs)
+        if self.allowed_extensions is None:
+            self.allowed_extensions = ['.json']
+        self.product_site_code = None
+        self.product_variables = None
+
+    def preprocess(self):
+        """Collect available input files and create the products, adding them to the collection to be published."""
+
+        # Don't want to publish the manifest file, so remove it from the collection
+        self.file_collection.pop()
+
+        # Read the manifest file and extract key parameters for product
+        with open(self.input_file) as f:
```

Comment:
Same here re: known file type, this validation could be moved to the resolve runner.

Reply:
Sure. The schema to validate against would still need to be defined here though, as it is specific to this handler.

Reply:
Ah, sorry. I was thinking of validating the content of the json, as below.

Reply:
Yeah was only thinking very high level stuff like "is it actually a valid JSON file". As you said, any further validation should be left up to the specific handler.